app/services/catalog/add_to_order.rb

```diff
@@ -19,12 +20,22 @@ def process
     private
 
     def order_item_params
-      @params.permit(:order_id, :portfolio_item_id, :count, :service_parameters => {}, :provider_control_parameters => {})
+      @params.permit(:order_id, :portfolio_item_id, :count, :provider_control_parameters => {})
+    end
+
+    def service_plan
+      @service_plan ||= Catalog::ServicePlans.new(@params[:portfolio_item_id]).process.items.first
     end
 
     def service_plan_ref
-      plans = Catalog::ServicePlans.new(@params[:portfolio_item_id]).process.items
-      plans.first["id"]
+      service_plan["id"]
+    end
+
+    def service_parameters
+      fields = service_plan["create_json_schema"].dig("schema", "fields").collect { |field| field["name"] }
+      @unfiltered_parameters.slice(*fields).permit!
```

Comment:
@lindgrenj6 
1. Do we have to check if the product has a modified parameter list, from the survey editing feature
2. Is this just checking if the parameter name matches or would it also check if all the constraints have been met.
3. How different is this from https://github.com/RedHatInsights/catalog-api/blob/cad9590cf9cba6969426dd307302d44b127d64bb/app/services/catalog/order_item_sanitized_parameters.rb#L2
4. What happens to encrypted fields do they travel in clear text to the Tower API (via HTTPS)?

Reply:
1. Yes, we can see if theres modified parameters, but since tower would do nothing with any extra ones the modified survey should contain the same keys, just with different constraints or default values. Either way the same keys always get sent, so lets store only the same keys.

2. Currently just checking the name matches so we only store the necessary variables. It doesn't check to see if the constraints have been met, the frontend does that. It is in scope but we haven't done it yet, eventually it will.

3. Now that I look at it it'd doing similar but it just runs before saving so we don't store any unnecessary garbage. Like if someone hit the API with bogus keys it only saves the ones that are pertinent to the survey.

4. The `OrderItemSanitizedParameters` class still runs and removes any passwords necessary before sending to approval - as far as I know when you order with a password field it still gets sent in plain text to topo - we don't do any encryption on either side. Is that something 

------
After chewing on this a bit, I think what I'm gonna do is refactor the OrderItemSanitizedParameters class to also include the field-filtering functionality like here, since it belongs there. 

Marking WIP...